apps/api/src/app/workflows-v2/usecases/sync-to-environment/sync-to-environment.usecase.ts

```diff
@@ -0,0 +1,167 @@
+          user: { ...command.user, environmentId: command.targetEnvironmentId },
+          identifierOrInternalId: externalId,
+        })
+      );
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  private async mapWorkflowToCreateWorkflowDto(
+    workflow: NotificationTemplateEntity,
+    preferences: PreferencesEntity[]
+  ): Promise<CreateWorkflowDto> {
+    return {
+      workflowId: workflow.triggers[0].identifier,
+      name: workflow.name,
+      active: workflow.active,
+      tags: workflow.tags,
+      description: workflow.description,
+      __source: WorkflowCreationSourceEnum.DASHBOARD,
+      steps: this.mapStepsToDto(workflow.steps),
+      preferences: this.mapPreferences(preferences),
+    };
+  }
+
+  private async mapWorkflowToUpdateWorkflowDto(
+    workflow: NotificationTemplateEntity,
+    existingWorkflow: NotificationTemplateEntity,
+    preferences: PreferencesEntity[]
+  ): Promise<UpdateWorkflowDto> {
+    return {
+      updatedAt: new Date().toISOString(),
+      workflowId: workflow.triggers[0].identifier,
+      name: workflow.name,
+      active: workflow.active,
+      tags: workflow.tags,
+      description: workflow.description,
+      steps: this.mapStepsToDto(workflow.steps, existingWorkflow.steps),
+      preferences: this.mapPreferences(preferences),
+    };
+  }
+
+  private mapStepsToDto(
+    steps: NotificationStepEntity[],
+    existingWorkflowSteps?: NotificationStepEntity[]
+  ): StepUpdateDto[] | StepCreateDto[] {
+    return steps.map((step) => ({
+      /*
+       * If we are updating an existing workflow, we need to map the updated steps to the existing steps
+       * (!) 'existingWorkflowSteps' are from a different environment than 'steps' - the only thing that doesn't change
+       *  in steps across environments is the stepId (TODO)
+       */
+      ...(existingWorkflowSteps && {
+        _id:
+          existingWorkflowSteps.find((existingStep) => existingStep.stepId === step.stepId)?._templateId ??
+          step._templateId,
+      }),
+      name: step.name ?? '',
+      type: step.template?.type ?? StepTypeEnum.TRIGGER,
+      controlValues: step.controlVariables ?? {},
+    }));
+  }
+
+  private mapPreferences(preferences: PreferencesEntity[]): {
+    user: WorkflowPreferences | null;
+    workflow: WorkflowPreferences | null;
+  } {
+    return {
+      user: preferences.find((pref) => pref.type === PreferencesTypeEnum.USER_WORKFLOW)?.preferences ?? null,
+      workflow: preferences.find((pref) => pref.type === PreferencesTypeEnum.WORKFLOW_RESOURCE)?.preferences ?? null,
+    };
+  }
+
+  private async getWorkflowPreferences(workflowId: string, environmentId: string): Promise<PreferencesEntity[]> {
+    const workflowPreferences = await this.preferencesRepository.find({
+      _templateId: workflowId,
+      _environmentId: environmentId,
+      type: {
+        $in: [PreferencesTypeEnum.WORKFLOW_RESOURCE, PreferencesTypeEnum.USER_WORKFLOW],
```

Comment:
question because i am missing the context, do we allow to `:workflowId/promote` code-first workflows as well?

Reply:
I think for now we agreed to **not** promote code-first workflows using dashboard UI, because they were some open questions and issues with that.